pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -95,6 +95,181 @@ func getApiURL(resourceOrWorkspace string, isAppInsightsQuery bool) string {
 	}
 }
 
+func buildLogAnalyticsQuery(query backend.DataQuery, dsInfo types.DatasourceInfo, appInsightsRegExp *regexp.Regexp) (*AzureLogAnalyticsQuery, error) {
+	queryJSONModel := types.LogJSONQuery{}
+	err := json.Unmarshal(query.JSON, &queryJSONModel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode the Azure Log Analytics query object from JSON: %w", err)
+	}
+	resources := []string{}
+	var resourceOrWorkspace string
+	var queryString string
+	resultFormat := dataquery.ResultFormat(types.TimeSeries)
+	appInsightsQuery := false
+	dashboardTime := false
+	timeColumn := ""
+	azureLogAnalyticsTarget := queryJSONModel.AzureLogAnalytics
+
+	if azureLogAnalyticsTarget.ResultFormat != nil {
+		resultFormat = *azureLogAnalyticsTarget.ResultFormat
+	}
+
+	// Legacy queries only specify a Workspace GUID, which we need to use the old workspace-centric
+	// API URL for, and newer queries specifying a resource URI should use resource-centric API.
+	// However, legacy workspace queries using a `workspaces()` template variable will be resolved
+	// to a resource URI, so they should use the new resource-centric.
+	if len(azureLogAnalyticsTarget.Resources) > 0 {
+		resources = azureLogAnalyticsTarget.Resources
+		resourceOrWorkspace = azureLogAnalyticsTarget.Resources[0]
+		appInsightsQuery = appInsightsRegExp.Match([]byte(resourceOrWorkspace))
+	} else if azureLogAnalyticsTarget.Resource != nil && *azureLogAnalyticsTarget.Resource != "" {
+		resources = []string{*azureLogAnalyticsTarget.Resource}
+		resourceOrWorkspace = *azureLogAnalyticsTarget.Resource
+	} else if azureLogAnalyticsTarget.Workspace != nil {
+		resourceOrWorkspace = *azureLogAnalyticsTarget.Workspace
+	}
+
+	if azureLogAnalyticsTarget.Query != nil {
+		queryString = *azureLogAnalyticsTarget.Query
+	}
+
+	if azureLogAnalyticsTarget.DashboardTime != nil {
+		dashboardTime = *azureLogAnalyticsTarget.DashboardTime
+		if dashboardTime {
+			if azureLogAnalyticsTarget.TimeColumn != nil {
```

Comment:
This could probably be pulled out into another function too, but this is the one I'd be least sure if it was worth the effort.

Reply:
Agreed on not being worth effort atm, maybe something for the feature if this property ends up being used elsewhere.